src/pages/network.tsx

```diff
@@ -0,0 +1,84 @@
+import { Input, Typography } from "@mui/material";
+import { useState } from "react";
+import { Divider } from "../components/common/Divider";
+import { OpaqueButton } from "../components/common/OpaqueButton";
+import { Row } from "../components/common/Row";
+import { useWallet } from "../contexts/wallet";
+import { useStore } from "../store/store";
+import theme from "../theme";
+
+export default function NetworkPage(){
+  const {walletAddress,getNetworkDetails} = useWallet();
+  const {network,setNetwork} = useStore((state) => state);
+  const [newNetworkRPCUrl,setNewNetworkRPCUrl] = useState<string>();
+  const [newNetworkPassphrase,setNewNetworkPassphrase] = useState<string>();
+
+  function fetchFromWallet(){
+    getNetworkDetails().then((networkDetails) => {
+      if(networkDetails.rpc){
+        setNewNetworkPassphrase(networkDetails.passphrase);
+        setNewNetworkRPCUrl(networkDetails.rpc);
+      }
+    })
+  }
+
+  function handleUpdatenetworkClick(){
+    if(newNetworkRPCUrl && newNetworkPassphrase){
+      setNetwork(newNetworkRPCUrl,newNetworkPassphrase);
```

Comment:
we should also reload the data here via `loadBlendData`, with the user's wallet address if it is connected.